src/frontend/apps/impress/src/features/left-panel/components/LeftPanel.tsx

```diff
@@ -2,34 +2,69 @@ import { Button } from '@openfun/cunningham-react';
 import { useRouter } from 'next/navigation';
 import { PropsWithChildren } from 'react';
 import { useTranslation } from 'react-i18next';
+import { css } from 'styled-components';
 
 import { Box, Icon, SeparatedSection } from '@/components';
+import { ButtonLogin } from '@/core';
 import { useCunninghamTheme } from '@/cunningham';
 import { useCreateDoc } from '@/features/docs';
 import { HEADER_HEIGHT } from '@/features/header/conf';
+import { LanguagePicker } from '@/features/language';
 import { useResponsiveStore } from '@/stores';
 
 export const LeftPanel = ({ children }: PropsWithChildren) => {
   const { t } = useTranslation();
   const router = useRouter();
-  const { isDesktop } = useResponsiveStore();
+  const { isDesktop, mobileMenuIsOpen, toggleMobileMenu } =
+    useResponsiveStore();
+
   const theme = useCunninghamTheme();
   const colors = theme.colorsTokens();
+  const spacings = theme.spacingsTokens();
 
   const { mutate: createDoc } = useCreateDoc({
     onSuccess: (doc) => {
       router.push(`/docs/${doc.id}`);
+      toggleMobileMenu();
     },
   });
 
   const goToHome = () => {
     router.push('/');
+    toggleMobileMenu();
   };
 
   const createNewDoc = () => {
     createDoc({ title: t('Untitled document') });
   };
 
+  const getContent = () => {
+    return (
+      <Box $width="100%">
+        <SeparatedSection>
+          <Box
+            $padding={{ horizontal: 'sm' }}
+            $width="100%"
+            $direction="row"
+            $justify="space-between"
+            $align="center"
+          >
+            <Box $direction="row" $gap="2px">
+              <Button
+                onClick={goToHome}
+                size="medium"
+                color="primary-text"
+                icon={<Icon iconName="house" />}
+              />
+            </Box>
+            <Button onClick={createNewDoc}>{t('New doc')}</Button>
+          </Box>
+        </SeparatedSection>
+        {children}
+      </Box>
+    );
+  };
+
```

Comment:
I find it is a bit anti pattern, what about creating a component for this part (`LeftPanelHeader` ?), you will be able to export part of the logic in this component moreover (createDoc / goToHome), making the component `LeftPanel` lighter, wdyt ?